File: CoffeeShop_Web/CoffeeShop/src/main/java/com/example/CoffeeShop/entity/Table.java
package com.example.CoffeeShop.entity;

import com.google.firebase.database.PropertyName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Table {
    private String id;
    private String status;
    private int tableNumber;
    private String createdAt;

    // THÊM BIẾN NÀY: Để đồng bộ với App Mobile
    private Integer mergedWith;

    private ActiveOrder currentOrder;

    @PropertyName("table_number")
    public int getTableNumber() {
        return tableNumber;
    }

    @PropertyName("table_number")
    public void setTableNumber(int tableNumber) {
        this.tableNumber = tableNumber;
    }

    @PropertyName("createdAt")
    public String getCreatedAt() {
        return createdAt;
    }

    @PropertyName("createdAt")
    public void setCreatedAt(String createdAt) {
        this.createdAt = createdAt;
    }

    // ÉP ĐỌC ĐÚNG TRƯỜNG "current_order" TỪ FIREBASE
    @PropertyName("current_order")
    public ActiveOrder getCurrentOrder() {
        return currentOrder;
    }

    @PropertyName("current_order")
    public void setCurrentOrder(ActiveOrder currentOrder) {
        this.currentOrder = currentOrder;
    }

    // --- BỔ SUNG: ÉP ĐỌC ĐÚNG TRƯỜNG "merged_with" TỪ FIREBASE ---
    @PropertyName("merged_with")
    public Integer getMergedWith() {
        return mergedWith;
    }

    @PropertyName("merged_with")
    public void setMergedWith(Integer mergedWith) {
        this.mergedWith = mergedWith;
    }
}

File: CoffeeShop_Moblie/app/src/main/java/com/example/coffeeshopmobile/data/model/Table.kt
package com.example.coffeeshopmobile.data.model

import com.google.firebase.database.IgnoreExtraProperties
import com.google.firebase.database.PropertyName

@IgnoreExtraProperties // <-- Lá chắn bảo vệ App không bị văng khi Firebase có dữ liệu lạ
data class Table(
    var id: String = "",

    @get:PropertyName("table_number")
    @set:PropertyName("table_number")
    var tableNumber: Int = 0,

    var status: String = "available",

    @get:PropertyName("merged_with")
    @set:PropertyName("merged_with")
    var mergedWith: Int? = null,

    var createdAt: String? = null,

    @get:PropertyName("current_order")
    @set:PropertyName("current_order")
    var currentOrder: ActiveOrder? = null
) {
    // ==========================================
    // CÁC HÀM PHỤ TRỢ (Giúp code UI cực kỳ nhàn)
    // ==========================================

    /**
     * Tự động cộng dồn tổng tiền của các món trong bàn
     */
    fun getTotalPrice(): Double {
        // Nếu currentOrder null thì trả về 0.0, nếu có thì tính tổng (giá * số lượng)
        return currentOrder?.items?.sumOf { it.price * it.quantity } ?: 0.0
    }

    /**
     * Lấy tổng số lượng ly nước khách đang gọi
     */
    fun getTotalItemCount(): Int {
        return currentOrder?.items?.sumOf { it.quantity } ?: 0
    }
}